src/woo_publications/contrib/documents_api/client.py

```diff
@@ -0,0 +1,138 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from datetime import date
+from uuid import UUID
+
+from django.http import HttpRequest
+
+from furl import furl
+from zgw_consumers.client import build_client
+from zgw_consumers.models import Service
+from zgw_consumers.nlx import NLXClient
+
+from .typing import EIOCreateBody, EIOCreateResponseBody
+
+__all__ = ["get_client"]
+
+
+def get_client(service: Service) -> DocumentenClient:
+    return build_client(service, client_factory=DocumentenClient)
+
+
+@dataclass
+class FilePart:
+    uuid: UUID
+    order: int
+    size: int
+
+
+@dataclass
+class Document:
+    uuid: UUID
+    lock: str
+    file_parts: list[FilePart]
+
+
+def _extract_uuid(url: str) -> UUID:
+    path = furl(url).path
+    last_part = path.segments[-1]
+    return UUID(last_part)
+
+
+class DocumentenClient(NLXClient):
+    """
+    Implement interactions with a Documenten API.
+
+    Requires Documenten API 1.1+ since we use the large file uploads mechanism.
+    """
+
+    def create_document(
+        self,
+        *,
+        identification: str,
+        source_organisation: str,
+        document_type_url: str,
+        creation_date: date,
+        title: str,
+        filesize: int,
+        filename: str,
+        author: str = "WOO registrations",
+        content_type: str = "application/octet-stream",
+        description: str = "",
+    ) -> Document:
+        data: EIOCreateBody = {
+            "identificatie": identification,
+            "bronorganisatie": source_organisation,
+            "informatieobjecttype": document_type_url,
+            "creatiedatum": creation_date.isoformat(),
+            "titel": title,
+            "auteur": author,
+            "status": "definitief",
```

Comment:
Can't we make a constant with all the statuses?

Reply:
There is a type definition for this already and the type checker enforces correctness. See https://github.com/GeneriekPublicatiePlatformWoo/registratie-component/pull/106/files#diff-b2726375446a03f8b580f3224284737684ea2b518e36b2afaf7c0b1ef7cfafc9R7 and https://github.com/GeneriekPublicatiePlatformWoo/registratie-component/pull/106/files#diff-b2726375446a03f8b580f3224284737684ea2b518e36b2afaf7c0b1ef7cfafc9R30

Try adding a typo this value and see what happens :)

Reply:
I see that Pyright complains when the typing is incorrect. So in that case looks good to me.